src/test/scala/is/hail/methods/KeyTableSuite.scala

```diff
@@ -413,4 +413,100 @@ class KeyTableSuite extends SparkSuite {
     val kt = hc.importTable("src/test/resources/sampleAnnotations.tsv", impute = true)
     assert(kt.same(kt))
   }
+
+  @Test def testUngroup() {
+    val data = Array(Array("Sample1", Row(23.0, "rabbit"), 9, 5, Row(1, "foo", 6.0), Row(7, "z", 10.0), Row(1, "foo", 6.0)))
+
+    val rdd = sc.parallelize(data.map(Row.fromSeq(_)))
+    val kt = KeyTable(hc, rdd, TStruct(
+      ("Sample", TString),
+      ("field0", TStruct(("1", TFloat32), ("2", TString))),
+      ("field1", TInt32),
+      ("field2", TInt32),
+      ("field3", TStruct(("a", TInt32), ("b", TString), ("c", TFloat64))),
+      ("field4", TStruct(("field0", TInt32), ("field1", TString), ("field2", TFloat64))),
+      ("field5", TStruct(("a", TInt32), ("b", TString), ("c", TFloat64)))), key = Array("Sample"))
+
+    assert(kt.ungroup(Array("field0")).signature == TStruct(("Sample", TString), ("1", TFloat32), ("2", TString),
+      ("field1", TInt32), ("field2", TInt32),
+      ("field3", TStruct(("a", TInt32), ("b", TString), ("c", TFloat64))),
+      ("field4", TStruct(("field0", TInt32), ("field1", TString), ("field2", TFloat64))),
+      ("field5", TStruct(("a", TInt32), ("b", TString), ("c", TFloat64)))))
+
+    assert(kt.ungroup(Array("field3")).signature == TStruct(("Sample", TString), ("field0", TStruct(("1", TFloat32), ("2", TString))),
+      ("field1", TInt32), ("field2", TInt32),
+      ("a", TInt32), ("b", TString), ("c", TFloat64),
+      ("field4", TStruct(("field0", TInt32), ("field1", TString), ("field2", TFloat64))),
+      ("field5", TStruct(("a", TInt32), ("b", TString), ("c", TFloat64)))))
+
+    val dataExpected = Array(Array("Sample1", 23.0, "rabbit", 9, 5, 1, "foo", 6.0, Row(7, "z", 10.0), Row(1, "foo", 6.0)))
+    val signatureExpected = TStruct(("Sample", TString), ("1", TFloat32), ("2", TString),
+      ("field1", TInt32), ("field2", TInt32),
+      ("a", TInt32), ("b", TString), ("c", TFloat64),
+      ("field4", TStruct(("field0", TInt32), ("field1", TString), ("field2", TFloat64))),
+      ("field5", TStruct(("a", TInt32), ("b", TString), ("c", TFloat64))))
+    val rddExpected = sc.parallelize(dataExpected.map(Row.fromSeq(_)))
+    val ktExpected = KeyTable(hc, rddExpected, signatureExpected)
+    assert(kt.ungroup(Array("field0", "field3")).same(ktExpected))
+
+    intercept[HailException](kt.ungroup(Array("field4")))
```

Comment:
use TestUtils.interceptFatal, and part of the message you expect. It's hard to see on just reading this which kind of failure is expected